src/Illuminate/Redis/Connections/PhpRedisConnection.php

```diff
@@ -91,19 +91,18 @@ public function spop($key, $count = null)
      * Add one or more members to a sorted set or update its score if it already exists.
      *
      * @param  string  $key
-     * @param  array  $membersAndScoresDictionary
+     * @param  mixed  $dictionary
      * @return int
      */
-    public function zadd($key, array $membersAndScoresDictionary)
+    public function zadd($key, ...$dictionary)
     {
-        $arguments = [];
-
-        foreach ($membersAndScoresDictionary as $score => $member) {
-            $arguments[] = $score;
-            $arguments[] = $member;
+        if (count($dictionary) === 1) {
+            $dictionary = call_user_func_array('array_merge', call_user_func_array('array_map',
+                [null, array_values($dictionary[0]), array_keys($dictionary[0])]
+            ));
```

Comment:
Is there a way to use a simpler approach that is easier to read on first glance?